pkg/dockercreds/access_checker_test.go

```diff
@@ -111,10 +105,20 @@ func testAccessChecker(t *testing.T, when spec.G, it spec.S) {
 				writer.WriteHeader(404)
 			})
 
-			tagName := fmt.Sprintf("%s/some/image:tag", server.URL[7:])
+			hasAccess, err := HasWriteAccess(testKeychain{}, tagName)
+			require.Error(t, err)
+			assert.False(t, hasAccess)
+		})
+
+		it("wraps unhandled server errors with a reasonable error message", func() {
+			handler.HandleFunc("/v2/", func(writer http.ResponseWriter, request *http.Request) {
+				writer.WriteHeader(500)
+			})
 
 			hasAccess, err := HasWriteAccess(testKeychain{}, tagName)
 			require.Error(t, err)
```

Comment:
I think there is a matcher called `EqualError` that allows one to also assert on the error message in the same require, but that is a minor nit.